doc/docs/master/how-tos/use-jupyterhub/index.md

```diff
@@ -0,0 +1,73 @@
+# Use JupyterHub with `python-pachyderm`
+
+This section describes how you can use the `python-pachyderm`
+client from within the JupyterHub UI.
+
+!!! note
+    You need to have Pachyderm and JupyterHub installed on the
+    same Kubernetes cluster as described in
+    [Deploy Pachyderm with JupyterHub](../../deploy-manage/deploy/deploy-pachyderm-jupyterhub.md).
+
+## Overview
+
+JupyterHub is a popular data science platform that enables users
+to quickly spin out multiple single-tenant Jupyter Notebook server instances.
+Jupyter Notebook provides a Python interactive development environment (IDE)
+that is convenient for data science projects. Because of the built-in
+rich-text support, visualizations, the easy-to-use web interface, many
+enterprise users prefer Jupyter Notebooks to the classic Terminal prompt.
+JupyterHub brings all the benefits of Jupyter Notebooks without the need
+to install or configure anything on user machines except for a web browser.
+
+[python-pachyderm](https://github.com/pachyderm/python-pachyderm) is an
+official Python client for Pachyderm. For Python developers who prefer to
+communicate with Pachyderm directly through the API, rather than by using
+the `pachctl` tool, `python-pachyderm` is the right choice.
+The [API Documentation](https://pachyderm.github.io/python-pachyderm/python_pachyderm.m.html)
+describes various API operations that you can execute to interact with
+Pachyderm.
+
+When you deployed JupyterHub by using our deployment script, `python-pachyderm`
+was installed in JupyterHub so that you can run API requests directly from
+your Jupyter Notebook.
+
+### Difference in Pipeline Creation Methods
+
+One of the most significant advantages of using `python-pachyderm`
+is that you can avoid rebuilding your Docker images
+every time you update your code.
+You can use the `create_python_pipeline` function to create a pipeline from
+within your JupyterHub Notebook and store your code locally rather than
+in a Docker container.
+
+If you want to create a pipeline from a Docker image or non-Python code, you
+can use the
+[create_pipeline](https://pachyderm.github.io/python-pachyderm/python_pachyderm.m.html#python_pachyderm.Client.create_pipeline)
+method. This is the method Pachyderm uses when you run `pachctl create
+pipeline` or create a pipeline in the Pachyderm UI. When you use
+`create_pipeline`, you need to build a new Docker image and push
+it to an image registry every
+time you update the code in your pipeline. Users that are less familiar
+with Docker might find this process a bit cumbersome.
+
+When you use `python-pachyderm`, in addition to the mentioned `create_pipeline` method,
+you can use the [create_python_pipeline](https://pachyderm.github.io/python-pachyderm/python_pachyderm.m.html#python_pachyderm.create_python_pipeline) function that does not require
+you to include your code in a Docker image and rebuild it each time you make
```

Comment:
You said this already above. It seem weird to me to talk about fancy create pipeline, then normal create pipeline, then go back to talk more details about fancy. Group things together better? 

e.g Start with: python-pach support the normal create_pipeline method that is available through the CLI, etc etc. 

But it also has an extra abstraction layer that is particularly nice for python users... intorduce fancy create pipeline...